### ISSUE TITLE
Notifications not working in foreground on iOS

### ISSUE THREAD

**Issue by Abdelrahman-Nabil** [zo0r/react-native-push-notification#1808]
# Bug

Remote Notifications does not work when app is in foreground on iOS only
Background: Android ✓ iOS ✓
Closed: Android ✓ iOS ✓
Foreground: Android ✓ iOS ✗

## Environment info

OS: macOS 10.15.7
    CPU: (8) x64 Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
    Memory: 397.86 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.15.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.3, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 26
      Build Tools: 26.0.1

Library versions tested: 6.1.3 & 7

## Steps To Reproduce
in a fresh project:
1. Install and set-up Firebase (core & messaging)
2. Install and set-up react-native-push-notification and @react-native-community/push-notification-ios

It used to work on older versions but I don't really recall which version in specific and I can't go back to older version due to other concerns

**Comment by Dallas62** [comment]
Hi @Abdelrahman-Nabil 

As far as i know, this answer has already been answered.
Also not that this is not the good github repository, please look at the iOS one.
https://github.com/react-native-push-notification-ios/push-notification-ios

Regards,

**Comment by eyalyoli** [comment]
It seems to be that we are all lost and we'll continue to be until this issue is well documented!
I'm using RN v0.63 and had the same issues that everyone is having here and in other 14506 issues on this repo and on [react-native-push-notification](https://github.com/zo0r/react-native-push-notification). The problem in the documentation lies in the integration between both libs.

**Tested with:**
react-native: 0.63
@react-native-community/push-notification-ios: 1.8.0
react-native-push-notification: 7.2.0
Using Firebase Cloud Messaging (FCM) for android & ios

**The solution:**
1. What @lowell-list-cambia suggested [here](https://github.com/react-native-push-notification-ios/push-notification-ios/issues/63#issuecomment-639096024) forces `onNotification` to trigger on foreground (which wasn't the case for ios - android worked fine).
2. If you wish to show a notification in the foreground, you should keep the line:
`completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);`
instead of:
`completionHandler(UNNotificationPresentationOptionNone);`
because it seems that using `UNNotificationPresentationOptionNone` hides locally triggered notifications too.

**Notes:**
1. The best way to show push notifications in the foreground is handling it manually by calling `PushNotification.localNotification(...)` inside `onNotification` - because only then you will have a choice if to show or not.
2. Watch out for a loop when triggering a local notification inside `onNotification` (it will re-trigger `onNotification`), I just used a local notification marker in the notification data and did what @robbycp said (removing irrelevant props from the notification).
3. Don't forget to define the `channeId` for android local notifications to work.

I'll PR an updated documentation.

**Comment by mirzahayat** [comment]
i think you have missing the configuration in AppDelegate.mm


try to check this snippet.

```
#import <UserNotifications/UserNotifications.h>
#import <RNCPushNotificationIOS.h>
#import <Firebase.h>
```

```
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
..............
  [FIRApp configure];
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
...........
  return YES;
}
```

```
////Called when a notification is delivered to a foreground app.
-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
{
  completionHandler(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge);
}
```

**Comment by sagarhudge-eaton** [comment]
@mirzahayat  I am using 
`"@react-native-community/push-notification-ios": "^1.11.0"` and `"react-native-push-notification": "^8.1.1"`

With below code snippet remote notifications are not visible when app is foreground on latest `ios 16.6` also `onNotification` is not working

     `-(void)userNotificationCenter:(UNUserNotificationCenter *)center willPresentNotification:(UNNotification *)notification 
         withCompletionHandler:(void (^)(UNNotificationPresentationOptions options))completionHandler
        {
         NSLog( @"foreground %@",notification ); 
  
      if (@available(iOS 14.0, *)) {
      NSLog( @"foreground in" );

        completionHandler(UNNotificationPresentationOptionSound | UNNotificationPresentationOptionBanner | 
         UNNotificationPresentationOptionBadge);
        } else {
         NSLog( @"foreground out" );

         completionHandler(UNNotificationPresentationOptionSound | UNNotificationPresentationOptionAlert | 
          UNNotificationPresentationOptionBadge);
        }
        }`

**Comment by httc-nam** [comment]
My app runs well on old ver. However, after I upgrade to RN 0.72.10, I face same issue.
- Packages:
"react native": 0.72.10
"react-native-push-notification": "^8.1.1",
"@react-native-community/push-notification-ios": "^1.11.0",
- Device Info:
IOS 17.4.2

Does Anyone can help this issue ?
